packages/vite/src/node/server/hmr.ts

```diff
@@ -44,7 +46,7 @@ export interface HotUpdateContext {
   modules: Array<EnvironmentModuleNode>
   read: () => string | Promise<string>
   server: ViteDevServer
-  environment: string
+  environment: ModuleExecutionEnvironment | BuildEnvironment
```

Comment:
Is it actually possible to have a build environment in a hot update?

Reply:
You're right, fixed 👍🏼